### PR TITLE
chore(pylint): Bump Pylint to 2.9.6

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

Bumping Pylint as recent versions address a number of false positives related to `TYPE_CHECKING`. See https://github.com/PyCQA/pylint/issues/3382 for more details. Sniffed out a couple of bugs in the process.

### TESTING INSTRUCTIONS

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue: https://github.com/apache/superset/issues/9953
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/core.py [hunk 1]
```diff
@@ -536,9 +536,10 @@ def get_all_table_names_in_schema(
             ]
         except Exception as ex:  # pylint: disable=broad-except
             logger.warning(ex)
+            return []
```
Comment (john-bodley) [inline superset/models/core.py:539]: Kind of surprised Mypy didn't find this.

File: superset/models/tags.py [hunk 2]
```diff
@@ -46,6 +46,7 @@ class TagTypes(enum.Enum):
     can find all their objects by querying for the tag `owner:alice`.
     """
 
+    # pylint: disable=invalid-name
```
Comment (john-bodley) [inline superset/models/tags.py:49]: I opted not the change these enums to uppercase as the string representation of the lowercase name is used throughout and I wanted to minimize the size of the PR.

File: superset/utils/core.py [hunk 3]
```diff
@@ -1228,8 +1228,8 @@ def get_main_metric_name(metrics: Sequence[Metric]) -> Optional[str]:
 def ensure_path_exists(path: str) -> None:
     try:
         os.makedirs(path)
-    except OSError as exc:
-        if not (os.path.isdir(path) and exc.errno == errno.EEXIST):
+    except OSError as ex:
```
Comment (john-bodley) [inline superset/utils/core.py:1231]: This fixes a potential bug as `exc` is defined globally.

File: superset/utils/pandas_postprocessing.py [hunk 4]
```diff
@@ -593,7 +593,7 @@ def geohash_encode(
         )
         return _append_columns(df, encode_df, {"geohash": geohash})
     except ValueError:
-        QueryObjectValidationError(_("Invalid longitude/latitude"))
+        raise QueryObjectValidationError(_("Invalid longitude/latitude"))
```
Comment (john-bodley) [inline superset/utils/pandas_postprocessing.py:596]: Fixes a potential bug as the exception was never raised.

File: superset/cli.py [hunk 5]
```diff
@@ -480,7 +482,10 @@ def import_dashboards(path: str, recursive: bool, username: str) -> None:
             files.extend(path_object.rglob("*.json"))
         if username is not None:
             g.user = security_manager.find_user(username=username)
-        contents = {path.name: open(path).read() for path in files}
+            contents = {}
+            for path_ in files:
+                with open(path_) as file:
+                    contents[path_.name] = file.read()
```
Comment (villebro) [inline superset/cli.py:488]: This is slightly unrelated, but as you're bumping the indentation, I wonder if we should attempt importing at all if `username is None`? And if so (=we assume there may be use cases where an anonymous user can import), shouldn't this added indentation be restored to its original place?

File: superset/connectors/sqla/models.py [hunk 6]
```diff
@@ -1145,7 +1143,7 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
         having_clause_and = []
 
         for flt in filter:  # type: ignore
-            if not all([flt.get(s) for s in ["col", "op"]]):
+            if not all(flt.get(s) for s in ["col", "op"]):
```
Comment (villebro) [inline superset/connectors/sqla/models.py:1146]: If `pylint` caught this this is a nice improvement 👍

File: superset/examples/flights.py [hunk 7]
```diff
@@ -38,13 +38,13 @@ def load_flights(only_metadata: bool = False, force: bool = False) -> None:
...
+        pdf["ds"] = (  # pylint: disable=unsupported-assignment-operation
             pdf.YEAR.map(str) + "-0" + pdf.MONTH.map(str) + "-0" + pdf.DAY.map(str)
         )
         pdf.ds = pd.to_datetime(pdf.ds)
-        del pdf["YEAR"]
-        del pdf["MONTH"]
-        del pdf["DAY"]
+        del pdf["YEAR"]  # pylint: disable=unsupported-delete-operation
+        del pdf["MONTH"]  # pylint: disable=unsupported-delete-operation
+        del pdf["DAY"]  # pylint: disable=unsupported-delete-operation
```
Comment (villebro) [inline superset/examples/flights.py:47]: Should we use `drop()` here instead?

File: superset/connectors/sqla/models.py (hunk 6 above)
Comment (john-bodley) [reply superset/connectors/sqla/models.py:1146]: I know many people find Pylint somewhat verbose and cumbersome, but the linting rules and checks are far superior than other linters in my opinion.

**Review by villebro — APPROVED** [review]
LGTM once CI passes

File: superset/models/schedules.py [hunk 8]
```diff
@@ -31,17 +31,20 @@
 
 
 class ScheduleType(str, enum.Enum):
+    # pylint: disable=invalid-name
```
Comment (john-bodley) [inline superset/models/schedules.py:34]: I opted not the change these enums (and others) to uppercase as the string representation of the lowercase name is used in the database and I wanted to minimize the size of the PR.

**Pull request merged**